web/src/features/time/HistoricalTimeHeader.tsx

```diff
@@ -35,7 +35,12 @@ const clamp = (date: number, offset: number) => {
     return '';
   }
 
-  return new Date(newDate).toISOString();
+  // Offset to 3 hours ago (taking into consideration that we're over the end datetime so actually -2).This mimics the logic on the backend in get_latest_complete_flowtraced_dt
+  const d = new Date(newDate);
+  d.setUTCHours(new Date().getUTCHours() - 2);
+  d.setUTCMinutes(0, 0, 0);
+
+  return d.toISOString();
```

Comment:
Do we have to hardcode this?

It will make it wrong in the future if we reduce the time delay on the app backend which is a goal.

Reply:
Of course - do you mind linking me to what you're referring to?

Reply:
Just the static -2 hour offset really. If we manage to reduce the app backend delay to say 1 hour this will be "wrong" again.

So if possible it would be best to try and determine this from the state files first and last values instead.

If that's too complicated and takes too much time then maybe this is fine for now though.

Reply:
User's may visit from specific datetimes so we don't know the latest flowtraced from the backend. It should effect to many users and the outcome would be them being jumped to realtime ~3 hours which the data will still be visible for in the charts.

Hopefully when we do real-time in the future we can revisit this code and remove the delay!

Reply:
That's a fair point re: users visiting specific datetimes. I got tunnel-visioned trying to ensure that we're actually hitting the cache when users visit, since we have an hour discrepancy between the frontend and the cache

Reply:
We could send the latest flow traced data through the meta endpoint if it's needed later.